examples/image_pixmap_chessboard.cpp

```diff
@@ -0,0 +1,70 @@
+/*=============================================================================
+   Copyright (c) 2016-2020 Joel de Guzman
+
+   Distributed under the MIT License (https://opensource.org/licenses/MIT)
+=============================================================================*/
+#include "app.hpp"
+
+using namespace cycfi::artist;
+
+bool constexpr is_little_endian() {
+   union {
+      uint32_t i;
+      char c[4];
+   } cmp = {0x01020304};
+   return !(cmp.c[0] == 1);
+}
+
+int constexpr rows = 8;
+int constexpr cols = 8;
+int constexpr square_side = 100;
+int constexpr square_area = square_side * square_side;
+int constexpr width = cols * square_side;
+int constexpr height = rows * square_side;
+size_t constexpr pix_buf_size = rows * cols * square_area;
+
+uint32_t constexpr white = 0xffffffff;
+// on little endian systems RGBA is formatted as ABGR for values
+std::function<uint32_t()> black = []() { return is_little_endian() ? 0xff000000 : 0x000000ff; };
```

Comment:
Come to think of it, infra/support.hpp should probably have a `to_little_endian(x)` and `to_big_endian(x)` set of overloads for different scalar types.

Reply:
Added a issue in `cycfi/infra`: https://github.com/cycfi/infra/issues/12